### PR TITLE
feat(jwt-plugin): Add support for EdDSA

### PR DESCRIPTION
**Pull request by 27ascii**
### Summary

This change adds support for the EdDSA JWT verification

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Martin Kölbel [martin.koelbel@mercedes-benz.com](mailto:martin.koelbel@mercedes-benz.com) on behalf of Mercedes-Benz Tech Innovation GmbH, [Provider Information](https://github.com/mercedes-benz/daimler-foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: kong/plugins/jwt/daos.lua [hunk 1]
```diff
@@ -55,6 +56,7 @@ return {
                                                      "^PS256$",
                                                      "^PS384$",
                                                      "^PS512$",
+                                                     "^EdDSA$",
```
Comment (27ascii) [inline kong/plugins/jwt/daos.lua:59]: This is not directly related to the EdDSA support, however I wonder why the pkey checks are not applied for ES256, ES384 and ES512. 

Wouldn't it make sense to apply this check for all algorithms and exclude the symmetric ones?
```
if_match = {
  not_match =  "^HS", 
},
```

**Review by hanshuebner — APPROVED** [review]
Thank you for the contribution!  The change looks good to me, but I'd like another approval from @jschmid1

File: kong/plugins/jwt/daos.lua (hunk 1 above)
Comment (jschmid1) [reply kong/plugins/jwt/daos.lua:59]: right, this list has grown over time and indeed it would make sense to apply pkey checks to all asymmetrical keys. I'll add this to my pile. Thanks for brining that up @27ascii

**Review by jschmid1 — APPROVED** [review]

**Comment by jschmid1** [comment]
Thank you for your contribution @27ascii
